site/content/posts/2023-08-09-v0.25.0-impersonation-proxy-with-external-certs.md

```diff
@@ -0,0 +1,337 @@
+---
+title: "Pinniped v0.25.0: With External Certificate Management for the Impersonation Proxy and more"
+slug: v0-25-0-external-cert-mgmt-impersonation-proxy
+date: 2023-08-09
+author: Joshua T. Casey
+image: https://images.unsplash.com/photo-1618075254460-429d47b887c7?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=2148&q=80
+excerpt: "With v0.25.0 you get external certificate management for the impersonation proxy, easier scheduling of the kube-cert-agent, and more"
+tags: ['Joshua T. Casey','Ryan Richard', 'Benjamin Petersen', 'release', 'kubernetes', 'pki', 'pinniped', 'tls', 'mtls', 'kind', 'contour', 'cert-manager']
+---
+
+![Friendly seal](https://images.unsplash.com/photo-1618075254460-429d47b887c7?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=2148&q=80)
+*Photo by [karlheinz_eckhardt Eckhardt](https://unsplash.com/@karlheinz_eckhardt) on [Unsplash](https://unsplash.com/s/photos/seal)*
+
+With Pinniped v0.25.0 you get the ability to configure an externally-generated certificate for Pinnniped Concierge's impersonation proxy to serve TLS.
+
+To read more on this feature, and the design decisions behind it, see the [proposal](https://github.com/vmware-tanzu/pinniped/tree/main/proposals/1547_impersonation-proxy-external-certs).
+To read more about the impersonation proxy, see the [docs](https://pinniped.dev/docs/reference/supported-clusters/#background).
+
+To see the feature in practice on a local kind cluster, follow these instructions.
+This will perform mTLS between your local client (kubectl and the pinniped CLI) and the impersonation proxy.
+
+The setup: Using a kind cluster, Contour as an ingress to the impersonation proxy, and `cert-manager` to generate a TLS serving cert.
+
+```shell
+Docker desktop v1.20.1
+Kind v0.20.0
+Contour v1.25.2
+Pinniped v0.25.0
+pinniped CLI v0.25.0 (https://pinniped.dev/docs/howto/install-cli/)
+cert-manager v1.12.3
+````
+
+Set up kind to run with Contour, using the example kind cluster configuration file provided by Contour.
+
+```shell
+$ wget https://raw.githubusercontent.com/projectcontour/contour/main/examples/kind/kind-expose-port.yaml
+$ kind create cluster \
+    --config kind-expose-port.yaml \
+    --name kind-with-contour \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+```
+
+Install Contour (see https://projectcontour.io/getting-started/ for more details).
```

Comment:
"Now we will install Contour to provide our kind cluster with an Ingress Controller.  We will later deploy a  Contour HTTPProxy in order to create an IP address that we can use to access the Impersonation Proxy.".

Reply:
Resolving locally...